shell/ptty_windows.go

```diff
@@ -11,6 +13,25 @@ import (
 	"github.com/gruntwork-io/terragrunt/options"
 )
 
+func PrepareConsole() {
+	enableVirtualTerminalProcessing(os.Stderr)
+	enableVirtualTerminalProcessing(os.Stdout)
+}
+
+func enableVirtualTerminalProcessing(file *os.File) {
+
+	var mode uint32
+	handle := windows.Handle(file.Fd())
+	if err := windows.GetConsoleMode(handle, &mode); err != nil {
+		fmt.Printf("failed to get console mode: %v\n", err)
```

Comment:
1. Should we really ignore these errors? Or return them and cause the Terragrunt command to fail? I can't tell if this is something optional or not.
1. If we are going to ignore these errors, we should log them, probably with debug logging (`util.Debugf`), using the logger in `terragrunt.Options`, which can be passed in here as an arg.

Reply:
Yeah, we can't just ignore them, because `mode` would not be set - I've added `return` there - I can only test it on my Windows 10 and I'm not sure how it will behave on other Windows versions

Reply:
The `return` helps to prevent the code from continuing to run... But should the actual error be returned all the way up the stack, causing Terragrunt to exit? Or are these errors safe to ignore and just mean less pretty terminal output?